### PR TITLE
The lines are not necessary

### PR DESCRIPTION
**Pull request by guy-maurel**
as pc never will been nullptr, at least newline.

### CODE REVIEW

File: src/align.cpp [hunk 1]
```diff
@@ -628,10 +628,7 @@ void align_preprocessor(void)
 
       // step past the 'define'
       pc = chunk_get_next_nc(pc);
-      if (pc == nullptr)
-      {
-         break;
-      }
+      // if (pc == nullptr) will never occurs, at least it is a newline
```
Comment (mwoehlke-kitware) [inline src/align.cpp]: (FYI, correct grammar would be "occur" :slightly_smiling_face:.)

That said, `echo -n '#define' | ./uncrustify` will trip the `if` you are removing here...

File: src/align.cpp [hunk 2]
```diff
@@ -654,10 +651,7 @@ void align_preprocessor(void)
 
       // step to the value past the close parenthesis or the macro name
       pc = chunk_get_next(pc);
-      if (pc == nullptr)
```
Comment (mwoehlke-kitware) [inline src/align.cpp]: ...and `echo -n '#define a' | ./uncrustify` will trip this one.

**Comment by mwoehlke-kitware** [comment]
...that's "be liberal in what you accept" if don't feel like clicking the link. In other words, prefer to defend against users feeding you garbage, because crashing is rarely TRTTD¹.

(¹ The Right Thing To Do.)

**Comment by guy-maurel** [comment]
Well! Thanks for the two examples. I have tried to find anything... but you are the ONE!
I'll modify the PR to add two malformed sources.
Should it be wise to introduced a new liste such as "malformed.test" 
to catch all such cases?

**Comment by guy-maurel** [comment]
In an Internet-Draft of 2017, Martin Thomson argues that Postel's robustness principle actually leads to a lack of robustness, including security.
https://tools.ietf.org/html/draft-thomson-postel-was-wrong-02

**Comment by mwoehlke-kitware** [comment]
> Should it be wise to introduced a new liste such as "malformed.test" to catch all such cases?

For now, I think that would be good; yes. I would also add a comment before the PP that the file *must not* end with a newline for the test to be effective, in order to at least try to minimize the chance that someone adds one inadvertently. (I still have half a notion to, at some point, fix the many files that are missing newlines...)

At some point (see #1910) I am hoping to improve the testing infrastructure for tests that are not formatting tests, which would probably include these. Having them in a separate `.test` will help make it more obvious that if/when we do get tests that are more concerned with just running uncrustify (and things like verifying expected errors/logging/parsing) than comparing formatted output that these should be ported to that sort of test.

**Comment by mwoehlke-kitware** [comment]
> In an Internet-Draft of 2017, Martin Thomson argues that Postel's robustness principle actually leads to a lack of robustness, including security. https://tools.ietf.org/html/draft-thomson-postel-was-wrong-02

Interesting read! That article seems to deal mainly with the consequences of trying to adapt to bad input rather than rejecting it, resulting in folks not caring that they're "doing it wrong". I usually refer to it more in the sense of [defensive programming](https://en.wikipedia.org/wiki/Defensive_programming), which is more "don't launch the missile if the CRC is malformed" than "don't bend over backwards to try to make sense of nonsense".

*If* I were to apply that here, I guess it would be an argument that the lines you were looking askance at should cause uncrustify to abort with an error rather than just shrugging and trying to continue on. That might be plausible for the first one, though less so for the second, since I'm not convinced such a source file would be useless.

File: tests/cli/output/help.txt [hunk 3]
```diff
@@ -69,6 +69,6 @@ Note: Use comments containing ' *INDENT-OFF*' and ' *INDENT-ON*' to disable
       processing of parts of the source file (these can be overridden with
       enable_processing_cmt and disable_processing_cmt).
 
-There are currently 650 options and minimal documentation.
+There are currently 651 options and minimal documentation.
```
Comment (mihaipopescu) [inline tests/cli/output/help.txt]: I have no idea how come two comments is adding a new option.

Comment (mwoehlke-kitware) [reply tests/cli/output/help.txt]: I assume this is borked in master, and no one has made a PR specifically to fix it...

Comment (gmaurel) [reply tests/cli/output/help.txt]: It is now OK.

**Pull request merged**